app/src/main/java/org/mozilla/fenix/HomeActivity.kt

```diff
@@ -785,6 +791,17 @@ open class HomeActivity : LocaleAwareAppCompatActivity(), NavHostActivity {
         }
     }
 
+    override fun getSystemService(name: String): Any? {
+        if (LAYOUT_INFLATER_SERVICE == name) {
+            if (inflater == null) {
+                // We need to clone the inflater the first time to make sure it is associated with the right context.
```

Comment:
Why can't we just pass in `this` to set it with the right context the first time? I don't think this comment really helps me understand why we need `cloneInContext` or why `baseContext` is needed.